src/impl.ml

```diff
@@ -141,14 +141,6 @@ let create common filename size parent =
   with Failure x ->
     `Error(true, x)
 
-let open_with_creat path rw =
-  ( if not(Sys.file_exists path) then begin
-    Lwt_unix.openfile path [ Unix.O_CREAT; Unix.O_RDONLY ] 0o0644 >>= fun fd ->
-    Lwt_unix.close fd
-    end else return () ) >>= fun () ->
-  Vhd_format_lwt.IO.openfile path rw
```

Comment:
I think I might have reverted too much of your patch here.
Might be better to unrevert it and use Lwt_unix.openfile path [ Unix.O_RDWR; Unix.O_CREAT ] 0o0644